pages/api/flows/index.js

```diff
@@ -1,48 +1,69 @@
 import dbConnect from "../../../lib/dbConnect";
 import Flow from "../../../models/Flow";
 import { getAllFlows } from "../../../services/flowService";
+import { unstable_getServerSession } from "next-auth/next";
+import { authOptions } from "../auth/[...nextauth]";
 
 export default async function handler(request, response) {
+  const session = await unstable_getServerSession(
+    request,
+    response,
+    authOptions
+  );
+
   await dbConnect();
 
   if (request.method === "GET") {
     const flows = await getAllFlows();
     return response.status(200).json(flows);
```

Comment:
The same applies here. Since the route is unprotected, I can get all the data from your MongoDB. Click [here](https://capstone-project-git-feat-user-login-cathrintruchan.vercel.app/api/flows) to see what I mean.